code/modules/power/power.dm

```diff
@@ -275,10 +275,15 @@
 //M is a mob who touched wire/whatever
 //power_source is a source of electricity, can be powercell, area, apc, cable, powernet or null
 //source is an object caused electrocuting (airlock, grille, etc)
+//siemens_coeff - layman's terms, conductivity
+//adj_check - set to only shock adjacent mobs to src (vendors, airlocks, etc.)
 //No animations will be performed by this proc.
-/proc/electrocute_mob(mob/living/carbon/M, power_source, obj/source, siemens_coeff = 1)
+/proc/electrocute_mob(mob/living/carbon/M, power_source, obj/source, siemens_coeff = 1, adj_check = FALSE)
 	if(istype(M.loc,/obj/mecha))
 		return 0	//feckin mechs are dumb
+	if(adj_check)
+		if(!in_range(src,M))
```

Comment:
because it uses `get_dist` the same way as `in_range` and all we really care about is distance? adjacency seems to be for clickcode as well, which makes me mistrust it. why overcomplicate?